### ISSUE TITLE
Everything is slowed down until all old messages are loaded up to the current one

### ISSUE THREAD

**Issue by slrslr** [oxen-io/session-desktop#2576]
Describe the bug:

When using session on multiple devices. On a session that is less used, messages accumulate. I only have 2 openGroup and up to 10 common conversations. But when I start a session and in openGroup there are both 200 messages, so you can not switch between conversations, you can not scroll to the end of the conversation, you have to wait and wait... You can't even write coherently, everything is slowed down until all old messages are loaded up to the current one.

Suggestion for future releases: sessionNodes could store the flag that was the last message read/sent. For example, only download 10 messages to other devices before this last one. Deeper history only when scrolling up by the user. This would make the session smoother.

Other person in Session says:
"sounds like a threading issue.  Tasks like loading messages should not be in the main thread where user-interaction is handled."
Another says:
"Having the same issue. Like I said before it needlessly loads every message when hitting the "return to bottom" button"
Yet another person:
"Why does it take 30 seconds to scroll to the bottom when there's 600 unread chats?"

Other information:

Device: [NTB ram 4GB]
OS: [Linux Mint 21/Cinnamon]
Session Version or Git commit hash: 1.10.3

**Comment by slrslr** [comment]
@WX4300 If it will repeat again it may give dev. more idea if you can find recently modified and accessed files:
https://www.ecosia.org/search?q=linus+find+recently+modified+files
And also check file sizes of it.
And check logs (Settings/Help/Report bug) during the issue.

**Comment by KeeJef** [comment]
Yeah loading lots of open group messages can definitely cause a big slowdown, we need to look into this further, we will be implementing cross platform read indicators which may keep a lid on this too.

**Comment by ianmacd** [comment]
> Yeah loading lots of open group messages can definitely cause a big slowdown, we need to look into this further, we will be implementing cross platform read indicators which may keep a lid on this too.

Did you say "cross platform read indicators"?

That's the holy grail, the missing link, the crock of gold at the end of the rainbow.

I hope the work for this is actually scheduled.

**Comment by KeeJef** [comment]
> Did you say "cross platform read indicators"?

Yes it should be included in the config message update early next year
